### DISCUSSION TITLE
Unable to create the storage directory (/var/www/html/project/var/cache/dev/profiler).

### DISCUSSION THREAD

**Discussion by mcg-pimcore** [pimcore/pimcore#10526] [Q&A / Support] [answered]
Hello,

Facing issue after installation of pimcore on linux centos 7 dev server.

Unable to create the storage directory (/var/www/html/project/var/cache/dev/profiler).

I have applied many solutions including this one

HTTPDUSER=$(ps axo user,comm | grep -E '[a]pache|[h]ttpd|[_]www|[w]ww-data|[n]ginx' | grep -v root | head -1 | cut -d\  -f1)
sudo setfacl -dR -m u:"$HTTPDUSER":rwX -m u:$(whoami):rwX var
sudo setfacl -R -m u:"$HTTPDUSER":rwX -m u:$(whoami):rwX var

To confirm if this is a permission issue I have given 777 to cache folder but still getting same issue.

I also tried to create this folder manually but after refresh the same error comes and folder deleted automatically.

**Comment by beheist** [comment]
This is a directory rights issue 99,9%. Please  check if you ran the commands correctly. You can also try "chmod -R 777 ./var" as a last resort solution (dont do that in prod though).

**Reply by mzcentric** [reply]
please use sudo chmod -R 777 /var/www/html/project/var/ command to give permission to var directory inside project folder

**Reply by mcg-pimcore** [reply]
No this is not a permission issue. I have mentioned in my problem that I tried with same solution, but no success.

**Comment by mcg-pimcore** [comment, marked as answer]
The issue is fixed. Issue was with SELinux setting. After make it from enforcing to disabled. Application started work. Thanks to @fashxp He suggest me the solution.

**Reply by fashxp** [reply]
thx for sharing!

**Comment by matrix1690** [comment]
Hi!
The following code fixed my problem.

`chcon -R -t httpd_sys_content_rw_t /var/www/`

**Comment by naveenmuthukumar2306** [comment]
{"success":false,"message":"Unable to create a directory at \/var\/www\/html\/public\/var\/assets. mkdir(): Permission denied"}
![Screenshot 2025-04-09 161446](https://github.com/user-attachments/assets/696a4851-f63b-4c5c-88a0-d3db8ebf285c)

when i try to upload file in pimcore i getting issue please explain how to fix this issue
